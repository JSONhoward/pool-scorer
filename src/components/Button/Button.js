import React from 'react'
import styled from 'styled-components'

const ButtonStyled = styled.button`
height: 3rem;
margin: 5px;
width: ${props => props.width || '12.5rem'};
font-family: 'Cabin', sans-serif;
font-size: ${props => props.font || '2rem'};
border-radius: 5px;
box-shadow: 1px 1px 5px black;
border: none;
background-color: white;
cursor: pointer;
`

const Button = ({text, handler, width, font}) => {
    return (
        <ButtonStyled font={font} onClick={handler} width={width}>
            {text}
        </ButtonStyled>
    )
}

export default Button
