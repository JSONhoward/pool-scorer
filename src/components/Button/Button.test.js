import React from 'react'
import {create} from 'react-test-renderer'
import {screen, render} from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Button from './Button'

describe('Button Tests', () => {
    test('Snapshot Test', () => {
        const tree = create(<Button />).toJSON()

        expect(tree).toMatchSnapshot()
    })

    test('Receives Props', () => {
        const props = {
            text: 'test text',
            height: 5,
            width: 5,
            font: '1rem',
            handler: jest.fn(() => console.log('test handler'))
        }

        const component = create(<Button {...props} />).root

        expect(component.props).toEqual(props)
    })

    test('Renders text prop', () => {
        render(<Button text={'test'} />)

        expect(screen.getByText('test')).toBeDefined()
    })
})