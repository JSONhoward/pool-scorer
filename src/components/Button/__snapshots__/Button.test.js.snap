// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Tests Snapshot Test 1`] = `
<button
  className="sc-AxjAm dwRJcX"
/>
`;
